domino/domino.py

```diff
@@ -74,15 +87,17 @@ def authenticate(self, api_key=None, auth_token=None, domino_token_file=None):
         self.request_manager = _HttpRequestManager(get_auth_by_type(api_key=api_key,
                                                                     auth_token=auth_token,
                                                                     domino_token_file=domino_token_file))
-
+    @_authenticate
```

Comment:
Thanks Alex. A few questions occurred to me. Have you considered the following?

- What happens if a user prefers to use the API token to authenticate instead of a token file?
- What if the user is using the client from an external node (i.e., not in the deployment)?
- Do we really want to insert a call to authenticate() at every invocation of every method?

Reply:
> Thanks Alex. A few questions occurred to me. Have you considered the following?
> 
> * What happens if a user prefers to use the API token to authenticate instead of a token file?

`domino_token_file` will be empty and the `_authenticate` decorator will do nothing except calling method it is decorating, so no changes to the current behavior.

> * What if the user is using the client from an external node (i.e., not in the deployment)?

I think the user should be responsible to maintain the `domino_token_file` in such case.

Here is the context for the changes in this PR: https://dominodatalab.atlassian.net/browse/DOM-31477

> * Do we really want to insert a call to authenticate() at every invocation of every method?

It's not 100% elegant, and we can avoid calling it let's say for a method requiring authentication calling another such method, but do you think it's really worth doing? Our "authentication" basically does only reading token from the file.